sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -750,18 +768,19 @@ case class HashAggregateExec(
       finishRegularHashMap
     }
 
+    outputFunc = generateResultFunction(ctx)
     val doAggFuncName = ctx.addNewFunction(doAgg,
       s"""
          |private void $doAgg() throws java.io.IOException {
          |  ${child.asInstanceOf[CodegenSupport].produce(ctx, this)}
+         |  $childrenConsumed = true;
```

Comment:
Before my change, All the records of the the dataset were processed and HashMap populated(+ spilt to sorter) at once.
With my change, all the records are not processed at once due to the change at line#878
Hence we need to know if all the records of the dataset is processed, otherwise process them.
This variable serves the purpose.